x/stakeibc/keeper/msg_server_clear_balance_test.go

```diff
@@ -0,0 +1,116 @@
+	// hz
+	zoneAddress := types.NewZoneAddress(HostChainId)
+	hostZone := stakeibctypes.HostZone{
+		ChainId:        HostChainId,
+		ConnectionId:   ibctesting.FirstConnectionID,
+		HostDenom:      Atom,
+		IBCDenom:       IbcAtom,
+		RedemptionRate: sdk.NewDec(1.0),
+		Address:        zoneAddress.String(),
+		FeeAccount: &stakeibctypes.ICAAccount{
+			Address: feeAddress,
+			Target:  stakeibctypes.ICAAccountType_FEE,
+		},
+	}
+
+	amount := uint64(1_000_000)
+
+	user := Account{
+		acc: s.TestAccs[0],
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+
+	return ClearBalanceTestCase{
+		initialState: ClearBalanceState{
+			feeChannel: Channel{
+				PortID:    icatypes.PortPrefix + feeAccountOwner,
+				ChannelID: feeChannelID,
+			},
+		},
+		validMsg: stakeibctypes.MsgClearBalance{
+			Creator: user.acc.String(),
+			ChainId: HostChainId,
+			Amount:  amount,
+			Channel: feeChannelID,
+		},
+	}
+}
+
+func (s *KeeperTestSuite) TestClearBalance_Successful() {
+	tc := s.SetupClearBalance()
+
+	// Get the sequence number before the ICA is submitted to confirm it incremented
+	feeChannel := tc.initialState.feeChannel
+	feePortId := feeChannel.PortID
+	feeChannelId := feeChannel.ChannelID
+
+	startSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx(), feePortId, feeChannelId)
+	s.Require().True(found, "sequence number not found before reinvestment")
+
+	_, err := s.GetMsgServer().ClearBalance(sdk.WrapSDKContext(s.Ctx()), &tc.validMsg)
+	s.Require().NoError(err, "balance clears")
+
+	// Confirm the sequence number was incremented
+	endSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx(), feePortId, feeChannelId)
+	s.Require().True(found, "sequence number not found after reinvestment")
+	s.Require().Equal(endSequence, startSequence+1, "sequence number after reinvestment")
+}
+
+func (s *KeeperTestSuite) TestClearBalance_HostChainMissing() {
+	tc := s.SetupClearBalance()
+	// remove the host zone
+	s.App.StakeibcKeeper.RemoveHostZone(s.Ctx(), HostChainId)
+	_, err := s.GetMsgServer().ClearBalance(sdk.WrapSDKContext(s.Ctx()), &tc.validMsg)
+	s.Require().EqualError(err, "chainId: GAIA: host zone not registered")
+}
+
+func (s *KeeperTestSuite) TestClearBalance_FeeAccountMissing() {
+	tc := s.SetupClearBalance()
+	// remove the fee account from the host zone
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), HostChainId)
+	s.Require().True(found, "host zone not found")
+	hostZone.FeeAccount = nil
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+	_, err := s.GetMsgServer().ClearBalance(sdk.WrapSDKContext(s.Ctx()), &tc.validMsg)
+	s.Require().EqualError(err, "chainId: GAIA: fee account is not registered")
+}
+
+func (s *KeeperTestSuite) TestClearBalance_ParseCoinError() {
+	tc := s.SetupClearBalance()
+	// set invalid denom on host zone
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), HostChainId)
+	s.Require().True(found, "host zone not found")
```

Comment:
for future reference, sometimes it can be kind of confusing if you put an error-like description for a require statement rather than a description of what you're checking (I accidentally do this a lot too). 

For instance, if this test failed, you'd see something like:
```
"host zone not found", expected: True, actual: False. 
```
Which can be a little confusing

I think a rule of thumb is: rather than describing what the check is confirming, describe the variable that's being checked
```suggestion
	s.Require().True(found, "host zone found")
```

Reply:
whoops, updated